pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -600,14 +600,17 @@ func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig,
 				logger.Infof("deployment for osd %d already exists. updating if needed", osd.ID)
 				if err = updateDeploymentAndWait(c.context, c.clusterInfo, dp, opconfig.OsdType, strconv.Itoa(osd.ID), c.spec.SkipUpgradeChecks, c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
 					logger.Errorf("failed to update osd deployment %d. %v", osd.ID, err)
+					// TODO: should orch status be updated here also?
 				}
 			} else {
 				// we failed to create job, update the orchestration status for this pvc
 				logger.Warningf("failed to create osd deployment for pvc %q, osd %v. %v", osdProps.pvc.ClaimName, osd, createErr)
+				// TODO: orch status is not updated for this PVC as stated above
 				continue
 			}
 		}
 
+		// TODO: why are we doing this a second time? it already happens above
```

Comment:
🤔 I don't know, but it feels unnecessary.

Reply:
Agreed, looks very much duplicated. Perhaps it is a bug from an osd refactor. This code has changed a lot in the last few years!